app/helpers/service_helper.rb

```diff
@@ -0,0 +1,50 @@
+module ServiceHelper
+  private
+
+  def custom_attr_fields(host, custom_attrs)
+    validated_fields = []
+    custom_attrs.each do |attr_name, fields|
+      fields = downcase_keys(fields)
```

Comment:
firlds = fields.transform_keys(&:downcase)

Reply:
This is the only method you need to add to app/services/teamdynamix_api.rb
Just make the yaml the same format as the payload so you don't need to transform it as much.

```yaml
:create:
  :Attributes:
  - name: custom attribute name:
    id: integer_id
    value: integer or string value
  - name: custom attribute dynamic value:
    id: integer_id
    value: "lorem ipsum #{host.host_attribute_name}"
```

```ruby
def evaluate_attributes(host, create_attrs)
  if create_attrs[:Attributes].present?
    create_attrs[:Attributes].each do |attribute| do
      attribute.transform_keys!(&:downcase)
      attribute['value'] = eval("\"#{attribute['value']}\"")
    end
  end
  create_attrs
end
```

Reply:
why are we down casing the keys here? From your past code the Keys were capital.

What should the keys be?

Reply:
We are downcasing the keys only because we are reading it as downcase in the next line. Keys for the payload could be either downcase or upcase, doesn't matter. Downcasing only to be in sync with the way we are reading it.

Reply:
If we don't have to restrict eval, the code example you provided would work perfectly, updating it such